lib/routelib/routelib.lua

```diff
@@ -909,20 +909,37 @@ local function route_failover_f(rctx, arg)
     end
 
     return function(r)
-        local res = nil
+        local result = nil
         for i=1, limit do
-            res = rctx:enqueue_and_wait(r, t[i])
+            local res = rctx:enqueue_and_wait(r, t[i])
             if i > 1 and s then
                 -- increment the retries counter
                 s(s_id, 1)
             end
-            if (miss == true and res:hit()) or (miss == false and res:ok()) then
+
+            -- process result
+            if res:hit() then
+                -- return hit
                 return res
+            elseif res:ok() then
```

Comment:
```lua
if res:hit() then
  return res
elseif res:ok()  then
  if miss == true then
    rmiss = res
  else
    return res
  end
else
  rerr = res
end

if rmiss then
  return rmiss
else
  return rerr
end
```

debating if this is more clear or not? I think it's also fewer branches?

thoughts?